app/controllers/works_controller.rb

```diff
@@ -0,0 +1,69 @@
+class WorksController < ApplicationController
+  def home
+    @homepage = true
+    @works = Work.all
+
+    @spotlight = Work.spotlight
+    @books = Work.top_ten("book")
+    @movies = Work.top_ten("movie")
+    @albums = Work.top_ten("album")
+  end
+
+  def index
+    @books = Work.popularity("book")
+    @movies = Work.popularity("movie")
+    @albums = Work.popularity("album")
+  end
+
+  def show
+    @work = Work.find_by(id: params[:id].to_i)
+
+    head :not_found unless @work
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new work_params
+    if @work.save
+      flash[:success] = "Item successfully added"
+      redirect_to root_path
+    else
+      flash.now[:success] = "Item could not be added"
```

Comment:
maybe instead of adding this message to `[:success]`, you could do a line something like this in the controller
```
flash[:error] = @work.errors.messages
```

and in your view (probably application.html.erb) the following code:
```
<% work.errors.messages.each do |name, errors| %>
  <% errors.each do |message| %>
    error name: <% name %>
    error message: <% message %>
  <% end %>
<% end %>
```

Of course this is just some inspiration code, you can form it to what you need it to be :)